### PR TITLE
[th2-4571] added maxBatchSizeInBytes to the EventBatcher class

### PR DESCRIPTION
**Pull request by Nikita-Smirnov-Exactpro**
* renamed maxBatchSize to maxBatchSizeInItems

### CODE REVIEW

File: src/main/kotlin/com/exactpro/th2/common/utils/event/EventBatcher.kt [hunk 1]
```diff
@@ -32,7 +32,8 @@ import kotlin.concurrent.withLock
  * for a group has elapsed or number of events in it has reached `maxBatchSize`.
  */
 class EventBatcher(
-    private val maxBatchSize: Int = 100,
+    private val maxBatchSizeInBytes: Long = 1_024 * 1_024,
+    private val maxBatchSizeInItems: Int = 100,
```
Comment (ivandruzhinin) [inline src/main/kotlin/com/exactpro/th2/common/utils/event/EventBatcher.kt:36]: Do we need to batch by size in items and in bytes at the same time? 
If not then instead maybe pass a function which will calculate new batch size and test it against `maxBatchSize`?

File: src/main/kotlin/com/exactpro/th2/common/utils/event/EventBatcher.kt [hunk 2]
```diff
@@ -50,24 +51,68 @@ class EventBatcher(
...
+    }
 
-        override fun close() = send()
+    companion object {
+        /**
+		* 4 - magic number
+		* 1 - protocol version
+		* 4 - message sizes
+		* Collapsed constant = 9
+	    */
```
Comment (ivandruzhinin) [inline src/main/kotlin/com/exactpro/th2/common/utils/event/EventBatcher.kt:89]: Please correct formatting

File: src/test/kotlin/com/exactpro/th2/common/utils/event/EventBatcherTest.kt [hunk 3]
```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright 2022 Exactpro (Exactpro Systems Limited)
```
Comment (ivandruzhinin) [inline src/test/kotlin/com/exactpro/th2/common/utils/event/EventBatcherTest.kt:2]:
```suggestion
 * Copyright 2023 Exactpro (Exactpro Systems Limited)
```

**Comment by Nikita-Smirnov-Exactpro** [comment]
This change is already merged into master

**Pull request closed without merging**